Source/Core/Core/NetPlayServer.cpp

```diff
@@ -1551,72 +1584,117 @@ void NetPlayServer::AbortGameStart()
 }
 
 // called from ---GUI--- thread
-bool NetPlayServer::SyncSaveData()
+bool NetPlayServer::CollectSaveSyncInfo(SaveSyncInfo* sync_info)
 {
-  // We're about to sync saves, so set m_saves_synced to false (waits to start game)
-  m_saves_synced = false;
-
-  m_save_data_synced_players = 0;
-
   u8 save_count = 0;
-
   for (ExpansionInterface::Slot slot : ExpansionInterface::MEMCARD_SLOTS)
   {
     if (m_settings.exi_device[slot] == ExpansionInterface::EXIDeviceType::MemoryCard ||
         Config::Get(Config::GetInfoForEXIDevice(slot)) ==
             ExpansionInterface::EXIDeviceType::MemoryCardFolder)
     {
-      save_count++;
+      ++save_count;
     }
   }
 
-  const auto game = m_dialog->FindGameFile(m_selected_game_identifier);
+  auto game = m_dialog->FindGameFile(m_selected_game_identifier);
   if (game == nullptr)
   {
     PanicAlertFmtT("Selected game doesn't exist in game list!");
     return false;
   }
 
-  bool wii_save = false;
+  bool has_wii_save = false;
+  std::unique_ptr<IOS::HLE::FS::FileSystem> configured_fs;
+  std::optional<std::vector<u8>> mii_data;
+  std::vector<std::pair<u64, WiiSave::StoragePointer>> wii_saves;
+  std::optional<DiscIO::Riivolution::SavegameRedirect> redirected_save;
   if (m_settings.savedata_load && (game->GetPlatform() == DiscIO::Platform::WiiDisc ||
                                    game->GetPlatform() == DiscIO::Platform::WiiWAD ||
                                    game->GetPlatform() == DiscIO::Platform::ELFOrDOL))
   {
-    wii_save = true;
-    save_count++;
-  }
+    has_wii_save = true;
+    ++save_count;
 
-  std::optional<DiscIO::Riivolution::SavegameRedirect> redirected_save;
-  if (wii_save && game->GetBlobType() == DiscIO::BlobType::MOD_DESCRIPTOR)
-  {
-    auto boot_params = BootParameters::GenerateFromFile(game->GetFilePath());
-    if (boot_params)
+    configured_fs = IOS::HLE::FS::MakeFileSystem(IOS::HLE::FS::Location::Configured);
+    if (m_settings.savedata_sync_all_wii)
+    {
+      IOS::HLE::Kernel ios;
+      for (const u64 title : ios.GetES()->GetInstalledTitles())
+      {
+        auto save = WiiSave::MakeNandStorage(configured_fs.get(), title);
+        wii_saves.push_back(std::make_pair(title, std::move(save)));
```

Comment:
I know this is existing code but might as well simplify this:

```suggestion
        wii_saves.emplace_back(title, std::move(save));
```